dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx

```diff
@@ -71,31 +72,34 @@ export function AppRepoForm(props: IAppRepoFormProps) {
   const [filterExclude, setFilterExclude] = useState(false);
   const [secret, setSecret] = useState<ISecret>();
   const [selectedImagePullSecret, setSelectedImagePullSecret] = useState("");
+  const [imagePullSecrets, setImagePullSecrets] = useState<string[]>([]);
   const [validated, setValidated] = useState(undefined as undefined | boolean);
 
-  useEffect(() => {
-    dispatch(actions.repos.fetchImagePullSecrets(namespace));
-  }, [dispatch, namespace]);
-
   const {
     repos: {
-      imagePullSecrets,
       errors: { create: createError, update: updateError, validate: validationError },
       validating,
     },
     config: { appVersion },
     clusters: { currentCluster },
   } = useSelector((state: IStoreState) => state);
 
+  useEffect(() => {
+    fetchImagePullSecrets(currentCluster, namespace);
+  }, [dispatch, namespace, currentCluster]);
+
+  async function fetchImagePullSecrets(cluster: string, repoNamespace: string) {
+    setImagePullSecrets(await Secret.getDockerConfigSecretNames(cluster, repoNamespace));
+  }
+
   useEffect(() => {
     // Select the pull secrets if they are already selected in the existing repo
-    imagePullSecrets.forEach(pullSecret => {
-      const secretName = pullSecret.metadata.name;
+    imagePullSecrets.forEach(secretName => {
       if (repo?.spec?.dockerRegistrySecrets?.some(s => s === secretName)) {
         setSelectedImagePullSecret(secretName);
       }
     });
-  }, [imagePullSecrets, repo, selectedImagePullSecret]);
+  }, [imagePullSecrets, repo]);
```

Comment:
I removed `selectedImagePullSecret` from the dependencies here, as it was causing a bug: every time you selected a different image pull secret, it would trigger this effect, which would set the selected image pull secret back to the one that matches the repos current reference, effectively stopping you from ever changing it.